source/plot/Axis.ooc

```diff
@@ -0,0 +1,176 @@
+			result = result & Shapes text(position + labelOffset, this label, this fontSize + 4, "middle")
+			result = result & "</g>\n" clone()
+			if (base >= pow(10, this precision - 1) || base <= pow(10, - this precision)) {
+				baseOffset := FloatPoint2D new(numberOffset x, - axisAreaSize y - (plotAreaSize y / plotAreaSize x) * axisAreaSize x / 2 + this fontSize / 2)
+				result = result & Shapes text(position + baseOffset, this getScientificPower(base), this fontSize, "end")
+			}
+
+			tickValue := this getFirstTickValue()
+			position y += translationToOrigo y - plotAreaSize y - scaling y * tickValue
+			while (tickValue <= this max) {
+				result = result & this getTickSVG(tickValue, base, position, numberOffset, rightTickMarkerStartOffset, tickMarkerEndOffset, "end")
+				tickValue += this tick
+				position y += - scaling y * tick
+			}
+		}
+		result = result & "</g>\n" clone()
+		result
+	}
+
+	getFirstTickValue: func -> Float {
+		result: Float
+		if ((this min < this tick && this min > 0.0f) || (this min > this tick && this min > 0.0f && this min < 1.0f) )
+			result = this min + (this tick - Float modulo(this min, this tick))
+		else
+			result = this min - Float modulo(this min, this tick)
+		result
+	}
+
+	getTickSVG: func (tickValue, base: Float, position, numberOffset, tickMarkerOnOtherSideOffset, tickMarkerEndOffset: FloatPoint2D, textAnchor: String) -> String {
+		result := "<g desc='" clone() & tickValue toString() & "'>\n" clone()
+		if (this gridOn)
+			result = result & Shapes line(position, position + tickMarkerOnOtherSideOffset, "grey", FloatPoint2D new(5,5))
+		result = result & Shapes line(position, position + tickMarkerEndOffset, "black")
+		result = result & Shapes line(position + tickMarkerOnOtherSideOffset, position + tickMarkerOnOtherSideOffset - tickMarkerEndOffset, "black")
+		tickValue = base >= pow(10, this precision - 1) || base <= pow(10, - this precision) ? (tickValue / base) : tickValue
+		tempTick := tickValue toString()
+		tempTickInt := tickValue as Int toString()
+		result = result & Shapes text(position + numberOffset, tickValue == floor(tickValue) ? tempTickInt : tempTick, this fontSize, textAnchor)
+		tempTick free()
+		tempTickInt free()
+		result = result & "</g>\n" clone()
+		result
+	}
+
+	roundEndpoints: func {
+		if (this roundAxisEndpoints) {
+			this min = this roundToValueDigits(this min, 2, false)
+			this max = this roundToValueDigits(this max, 2, true)
+		}
+	}
+
+	roundToValueDigits: func (value: Float, valueDigits: Int, up: Bool) -> Float {
+		valueAndPower := this decomposeToValueAndPower(value, valueDigits)
+		result := valueAndPower[0]
+		if (valueAndPower[0] != 0) {
+			result = up ? ceil(valueAndPower[0]) : floor(valueAndPower[0])
+			result *= valueAndPower[1]
+		}
+		gc_free(valueAndPower data)
+		result
+	}
```

Comment:
Pretty sure this function should be in `FloatExtension.ooc`, not particular to the Plot library.